build.assets/build-arch.sh

```diff
@@ -0,0 +1,149 @@
+
+# Switch depending on the architecture which is passed as the script argument
+case $1 in
+    "arm64")
+        export ARCH="arm64"
+        export GO_ARCH="arm64"
+        export SYSROOT="${HOME}/x-tools/aarch64-centos7-linux-gnu/aarch64-centos7-linux-gnu/sysroot"
+
+        export PATH="${HOME}/x-tools/aarch64-centos7-linux-gnu/bin:$PATH"
+        export CC="aarch64-centos7-linux-gnu-cc --sysroot=${SYSROOT} -I${SYSROOT}/include" # Hacky but works
+        export CXX="aarch64-centos7-linux-gnu-c++ --sysroot=${SYSROOT}"
+        export LD="aarch64-centos7-linux-gnu-ld --sysroot=${SYSROOT}"
+        export PKG_CONFIG_PATH="${SYSROOT}/lib/pkgconfig"
+        ;;
+    "arm")
+        export ARCH="arm"
+        export GO_ARCH="arm"
+        export SYSROOT="${HOME}/x-tools/armv7-centos7-linux-gnueabi/armv7-centos7-linux-gnueabi/sysroot"
+
+        export PATH="${HOME}/x-tools/armv7-centos7-linux-gnueabi/bin:$PATH"
+        export CC="armv7-centos7-linux-gnueabi-cc --sysroot=${SYSROOT} -I${SYSROOT}/include -fPIC" # Hacky but works
+        export CXX="armv7-centos7-linux-gnueabi-c++ --sysroot=${SYSROOT}"
+        export LD="armv7-centos7-linux-gnueabi-ld --sysroot=${SYSROOT}"
+        export PKG_CONFIG_PATH="${SYSROOT}/lib/pkgconfig"
+        ;;
+    "386")
+        export ARCH="i686"
+        export GO_ARCH="386"
+        export SYSROOT="${HOME}/x-tools/i686-centos7-linux-gnu/i686-centos7-linux-gnu/sysroot"
+
+        export PATH="${HOME}/x-tools/i686-centos7-linux-gnu/bin:$PATH"
+        export CC="i686-centos7-linux-gnu-cc --sysroot=${SYSROOT} -I${SYSROOT}/include" # Hacky but works
+        export CXX="i686-centos7-linux-gnu-c++ --sysroot=${SYSROOT}"
+        export LD="i686-centos7-linux-gnu-ld --sysroot=${SYSROOT}"
+        export PKG_CONFIG_PATH="${SYSROOT}/lib/pkgconfig"
+        ;;
+    "amd64")
+        export ARCH="amd64"
+        export GO_ARCH="amd64"
+        export SYSROOT="${HOME}/x-tools/x86_64-centos7-linux-gnu/x86_64-centos7-linux-gnu/sysroot"
+
+        export PATH="${HOME}/x-tools/x86_64-centos7-linux-gnu/bin:$PATH"
+        export CC="x86_64-centos7-linux-gnu-cc --sysroot=${SYSROOT} -I${SYSROOT}/include" # Hacky but works
+        export CXX="x86_64-centos7-linux-gnu-c++ --sysroot=${SYSROOT}"
+        export LD="x86_64-centos7-linux-gnu-ld --sysroot=${SYSROOT}"
+        export PKG_CONFIG_PATH="${SYSROOT}/lib/pkgconfig"
+        ;;
+    *)
+        echo "Unknown architecture $1"
+        exit 1
+        ;;
+esac
+
+
+function build_3rdparty() {
+rm -rf 3rdparty-${ARCH}
+mkdir -p 3rdparty-${ARCH}
+
+cd 3rdparty-${ARCH}
+
+# Unlock sysroot
+chmod -R +w "${SYSROOT}"
+
+# Build and install
+
+#zlib
+git clone https://github.com/madler/zlib.git
+cd zlib
+./configure --prefix="${SYSROOT}"
+make -j$(nproc)
+make install
+
+cd ..
+
+#libzstd
+git clone https://github.com/facebook/zstd.git
+cd zstd
+
+make -j$(nproc)
+make install PREFIX=${SYSROOT}
+
+cd ..
+
+#libelf
+git clone https://github.com/arachsys/libelf.git
```

Comment:
Needs to be pinned to a commit SHA before building.

Reply:
https://github.com/gravitational/teleport/pull/36066#discussion_r1437267058